src/redux/features/observations/observationsSelectors.ts

```diff
@@ -55,7 +55,22 @@ export const selectMergedPlantingSiteObservations = createCachedSelector(
   }
 )((state: RootState, plantingSiteId: number, defaultTimeZone: string) => `${plantingSiteId}_${defaultTimeZone}`); // planting site id / default time zone is the key for the cache
 
-// add more selectors for drill down views, as needed
+// search observations (search planting zone name only)
+export const searchObservations = createCachedSelector(
+  (state: RootState, plantingSiteId: number, defaultTimeZone: string, search: string) =>
+    selectMergedPlantingSiteObservations(state, plantingSiteId, defaultTimeZone),
+  (state: RootState, plantingSiteId: number, defaultTimeZone: string, search: string) => search,
+  (observations, search) =>
+    observations?.filter((observation: ObservationResults) => {
+      const regexp = new RegExp(search, 'i');
```

Comment:
This should allow regular expressions to be input into the search input and ran unescaped against the zone names. Other searches use the search API, making this behavior different. Perhaps not worth worrying about, but just noting.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_expressions#escaping has a function that could be used if escaping is desired here.

Reply:
I can certainly add the escaping 👍 , fwiw i didn't know regex search was a requirement in our product. you never know, we could have a tech savvy customer..

Reply:
btw, I was going to use "string.toLowerCase().includes(pattern.toLowerCase())" but went with regexp purely for the case-insensitive option. I think I understand the reason for the suggestion now.
If it's less confusing, I can switch back to "string.includes" approach.

Reply:
Oh, the regexp approach isn't confusing. I just wondered whether allowing full regexp searches was what we wanted here. I'm okay with either approach. Regexp with escaping vs. string.includes should give the same result (I _think_), and either one is clear enough